cmd/frostfs-node/cache.go

```diff
@@ -26,7 +26,9 @@ type valueWithTime[V any] struct {
 
 // entity that provides TTL cache interface.
 type ttlNetCache[K comparable, V any] struct {
-	ttl time.Duration
+	m       sync.RWMutex        // protects progMap
+	progMap map[K]chan struct{} // contains fetch-in-progress keys
```

Comment:
We already have `valueWithTime` which can be extended. Can we avoid having a map and add a mutex to `valueWithTime` instead?

Reply:
what if multiple routines come and request the same value?

Reply:
I think it's about something like this:
map[key]{*value, keyMutex}

First get object from this map with global mutex (it can be optimized by using hashed key and shards), then lock keyMutex to get value.

Reply:
I think it's about something like this:
map[key]{*value, keyMutex}

First get object from this map with global mutex (it can be optimized by using hashed key and shards), then lock keyMutex to get value.